### ISSUE TITLE
Analyze Documentation generation via automation

### ISSUE THREAD

**Issue by neontvn** [cse210-fa24-group13/codesnip#7]
Possible Option: JSDocs

**Comment by neontvn** [comment]
- TypeDoc for typescript code 
- JSDocs seems a better option for the web application in js


**Unknowns**
- How to setup JSDoc and TypeDoc together in the same repo?

**Comment by Kshitijde** [comment]
better-docs allows document generation for both JS and TS files. This has been tested with sample files and documents were generated satisfactorily well. 

A caveat though is that a particular format of descriptions and comments needs to be maintained to achieve this. The current code base does not adhere to these requirements and we would need to make modifications in that regard to generate decent documentation.

In conclusion, we have decided to go ahead with better-docs for now, certain formatting choices need to be kept in mind.
